app/src/examples/RuntimeTests/tests/core/useAnimatedStyle/reuseAnimatedStyle.test.tsx

```diff
@@ -0,0 +1,144 @@
+      </View>
+    );
+  };
+
+  const TEST_CASES: Array<{ startStyle: AnimatableValueObject; finalStyle: AnimatableValueObject; animate: boolean }> =
+    [
+      { startStyle: { width: 100 }, finalStyle: { width: 300 }, animate: true },
+      { startStyle: { left: 10 }, finalStyle: { left: 30 }, animate: true },
+      { startStyle: { top: 10 }, finalStyle: { top: 30 }, animate: true },
+      { startStyle: { opacity: 0 }, finalStyle: { opacity: 1 }, animate: true },
+      { startStyle: { height: 20 }, finalStyle: { height: 100 }, animate: true },
+      { startStyle: { margin: 0 }, finalStyle: { margin: 100 }, animate: true },
+      { startStyle: { margin: -100 }, finalStyle: { margin: 0 }, animate: true },
+      { startStyle: { opacity: 0 }, finalStyle: { opacity: 1 }, animate: false },
+      { startStyle: { height: 20 }, finalStyle: { height: 100 }, animate: false },
+      { startStyle: { margin: 0 }, finalStyle: { margin: 100 }, animate: false },
+      { startStyle: { margin: -100 }, finalStyle: { margin: 0 }, animate: false },
+      { startStyle: { margin: -50, height: 20 }, finalStyle: { margin: 40, height: 200 }, animate: true },
+      { startStyle: { margin: 50, height: 20 }, finalStyle: { margin: -40, height: 200 }, animate: true },
+      { startStyle: { margin: -50, top: 20 }, finalStyle: { margin: 40, top: 200 }, animate: true },
+      {
+        startStyle: { margin: -50, top: 20, height: 20 },
+        finalStyle: { margin: 40, top: 200, height: 200 },
+        animate: true,
+      },
+      { startStyle: { height: 20, width: 100 }, finalStyle: { height: 100, width: 300 }, animate: true },
+      {
+        startStyle: { height: 20, width: 100, backgroundColor: 'white' },
+        finalStyle: { height: 100, width: 300, backgroundColor: 'royalblue' },
+        animate: true,
+      },
+      {
+        startStyle: { height: 20, width: 100, margin: 0, backgroundColor: 'white' },
+        finalStyle: { height: 100, width: 300, margin: 20, backgroundColor: 'royalblue' },
+        animate: true,
+      },
+    ];
+
+  test.each(TEST_CASES)(
+    'Animate 3 components from ${startStyle} to ${finalStyle}, animate=${animate}',
+    async ({ startStyle, finalStyle, animate }) => {
+      await render(<TripleComponent startStyle={startStyle} finalStyle={finalStyle} animate={animate} />);
+      const componentOne = getTestComponent(COMPONENT_REF.ONE);
+      const componentTwo = getTestComponent(COMPONENT_REF.TWO);
+      const componentThree = getTestComponent(COMPONENT_REF.THREE);
+
+      await wait(1000);
+
+      // Check the distance from the top
+      const componentHeight = 'height' in finalStyle ? (finalStyle.height as number) : 80;
+      const componentMargin = 'margin' in finalStyle ? (finalStyle.margin as number) : 0;
+      const componentTop = 'top' in finalStyle ? (finalStyle.top as number) : 0;
```

Comment:
That's great that you're checking it! Could we reduce the number of `if conditions` in the test? How about moving the margin/position checks to a separate test case?

Reply:
> How about moving the margin/position checks to a separate test case?
I think I've found an easier solution